### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by omar25ahmed**
# Learning Objectives:
- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical questions.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,3 +5,69 @@ VALUES ('Agumon', '2020-02-03', 0, 'true', 10.23),
...
+  -- Begin the transaction
+      BEGIN;
+  -- Set the condition (update the species column Where the name ends with 'mon')
+    UPDATE animals
+    SET species = 'digimon'
+    WHERE name LIKE '%mon';
+  -- Set the condition (update the species column Where the name doesn't end with 'mon')
+    UPDATE animals
+    SET species = 'pokemon'
+    WHERE name NOT LIKE '%mon';
```
Comment (keroloussamy) [inline data.sql]: - [ ] Kindly here the requirements say `Update the animals table by setting the species column to pokemon for all animals that don't have species already set.` So you have to check if it's equal to NULL.

File: data.sql [hunk 2]
```diff
@@ -5,3 +5,69 @@ VALUES ('Agumon', '2020-02-03', 0, 'true', 10.23),
...
+    SAVEPOINT delete_animals;
+  -- make the weight to be negative
+    UPDATE animals SET weight_kg = weight_kg * (-1);
+  -- Rollback to the first savepoint
+    ROLLBACK TO delete_animals;
+  -- Modify the weight to be all positive
+    UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+  -- Then commit that changes
+    COMMIT;
+
```
Comment (keroloussamy) [inline data.sql]: - [ ] Kindly move all these transactions to `queries.sql` file, the `data.sql` file for inserting data.

File: data.sql [hunk 3]
```diff
@@ -5,3 +5,69 @@ VALUES ('Agumon', '2020-02-03', 0, 'true', 10.23),
...
+      ROLLBACK;
+    -- Check if the table exists
+      SELECT EXISTS (SELECT FROM pg_tables WHERE schemaname = 'public' AND tablename  = 'animals');
+      -- return true
+
+-- Fourth Transaction
+  -- Begin the transaction
+    BEGIN;
+  -- Delete the animals where got born after 2022
+    WHERE date_of_birth >= '2022-01-01';
```
Comment (keroloussamy) [inline data.sql]: - [ ] Kindly here the requirements say `Delete all animals born after Jan 1st, 2022.` so you have written the condition but you didn't write the `delete` statement.

File: queries.sql [hunk 4]
```diff
@@ -17,3 +17,32 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+
+-- How many animals have never tried to escape?
+SELECT COUNT(escape_attempts)
+FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT COUNT(escape_attempts)
+FROM animals
+WHERE escape_attempts = 0;
```
Comment (keroloussamy) [inline queries.sql]: - [ ] Kindly here you have to calculate the average weight of animals, not the number of animals has never tried to escape, you made this query already above.

File: queries.sql [hunk 5]
```diff
@@ -17,3 +17,32 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+-- How many animals are there? 10
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+
```
Comment (keroloussamy) [inline queries.sql]: - [ ] Kindly you have missed the `How many animals are there?` query, you have instead this update query, Kindly add it.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @omar25ahmed, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @omar25ahmed , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**